src/v/cloud_storage/cache_service.h

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/vectorizedio/redpanda/blob/master/licenses/rcl.md
+ */
+
+#pragma once
+
+#include "seastarx.h"
+
+#include <seastar/core/future.hh>
+#include <seastar/core/iostream.hh>
+
+#include <filesystem>
+
+struct cache_item {
+    ss::input_stream<char> body;
+    uint64_t size;
+};
+
+class cache {
+public:
+    /// C-tor.
+    ///
+    /// \param cache_dir is a directory where cached data is stored
+    /// \param max_cache_size is a maximum size of the cached data on disk
+    /// \param cache_period is a time interval used to check if cache should be
+    /// cleaned
+    cache(
+      std::filesystem::path cache_dir,
+      uint64_t max_cache_size,
```

Comment:
`size_t`

Reply:
fixed! just so I understand: are there some situations where we use `uint32_t` or `uint64_t`? Or we generally prefer `size_t`?

Reply:
the types like `int32_t` or `uint64_t` are so called fixed-size types because the size is right there in the name and they are guaranteed to have that particular size. the guarantees that other types make may not be tied to size. for example `size_t` is `size_t can store the maximum size of a theoretically possible object of any type (including array).` according to this page https://en.cppreference.com/w/c/types/size_t and my interpretation of that is that on 32-bit machine size_t could be 32 bits, but on 64 bit machines it is 64 bits. I don't know all the details, and that might be an incorrect assessment of size_t, but the point is that semantics from the name of the type the size isn't necessarily fixed.

this is all relevant when we want to store data on disk or send data over the wire. For example, if we went an `int` over the wire one machine may use `32 bits` for the integer while another machine might use `64 bits`. it isn't common because now days everything is 64 bits, but it does help eliminate bugs and it is still possible as systems become heterogenous.

Reply:
got it, looks like we try to use `size_t` where semantically makes sense. Thanks!

Reply:
@dotnwat I personally am used to the convention that `size_t` is only used for the size of objects in memory and file sizes and offsets is something else (`uint64_t` or perhaps a typedef). So using `uint64_t` for the cache size makes sense to me.
* It helps with portability. Of course we aren't going to port redpanda to 32bit smart lightbulbs anytime soon, but it is still a nice to have. I remember when someone tried to compile ClickHouse on a 32bit system, he had to make a lot of small changes substituting `size_t` for a 64bit type.
* It has some small documentation value - reader of the code knows which one is a memory size and which one is an on-disk size.
* It is also the convention that seastar uses - e.g. in `file::dma_write` file position is `uint64_t`, but buffer size is `size_t`.

Reply:
I don't disagree with anything you're saying, but I think it is valuable to maintain some consistency. And that generally means (with some exceptions of course) using signed integers, and fixed size integers in serialization contexts.

This particular topic hasn't come up in a long time, but I think it is worth revisiting as a group, converging on a standard or some guidelines, and getting that written down some where. Want to put that on the agenda for core panda next week?

In the mean time it really doesn't matter much size_t vs uint64_t or whatever from a correctness perspective. So I think its fine int his PR to use whatever you want :)